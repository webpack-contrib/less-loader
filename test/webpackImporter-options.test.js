import {
  compile,
  getCodeFromBundle,
  getCodeFromLess,
  getCompiler,
  getErrors,
  getWarnings,
} from "./helpers";

describe('"webpackImporter" option', () => {
  it("should work when value is not specify", async () => {
    const testId = "./import-webpack.less";
    const compiler = getCompiler(testId);
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId);

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work when value is "true"', async () => {
    const testId = "./import-webpack.less";
    const compiler = getCompiler(testId, {
      webpackImporter: true,
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId);

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work when value is "false"', async () => {
    const testId = "./import.less";
    const compiler = getCompiler(testId, {
      webpackImporter: false,
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId);

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should throw an error on webpack import when value is "false"', async () => {
    const testId = "./import-webpack.less";
    const compiler = getCompiler(testId, {
      webpackImporter: false,
    });
    const stats = await compile(compiler);

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
