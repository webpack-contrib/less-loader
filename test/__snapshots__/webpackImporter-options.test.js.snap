// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"webpackImporter" option should throw an error on webpack import when value is "false": errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
",
]
`;

exports[`"webpackImporter" option should throw an error on webpack import when value is "false": warnings 1`] = `Array []`;

exports[`"webpackImporter" option should work when value is "false": css 1`] = `
"@import \\"css.css\\";
@import \\"css.css\\";
.classical-css,
#it-works {
  background: hotpink;
}
.box,
#it-works {
  color: #fe33ac;
  border-color: #fdcdea;
  background: url(box.png);
}
.box div {
  -webkit-box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
}
#it-works {
  margin: 10px;
}
"
`;

exports[`"webpackImporter" option should work when value is "false": errors 1`] = `Array []`;

exports[`"webpackImporter" option should work when value is "false": warnings 1`] = `Array []`;

exports[`"webpackImporter" option should work when value is "true": css 1`] = `
"@import \\"some/css.css\\";
@import \\"some/css.css\\";
#it-works {
  color: hotpink;
}
.modules-dir-some-module,
#it-works {
  background: hotpink;
}
#it-works {
  margin: 10px;
}
"
`;

exports[`"webpackImporter" option should work when value is "true": errors 1`] = `Array []`;

exports[`"webpackImporter" option should work when value is "true": warnings 1`] = `Array []`;

exports[`"webpackImporter" option should work when value is not specify: css 1`] = `
"@import \\"some/css.css\\";
@import \\"some/css.css\\";
#it-works {
  color: hotpink;
}
.modules-dir-some-module,
#it-works {
  background: hotpink;
}
#it-works {
  margin: 10px;
}
"
`;

exports[`"webpackImporter" option should work when value is not specify: errors 1`] = `Array []`;

exports[`"webpackImporter" option should work when value is not specify: warnings 1`] = `Array []`;
